Components/ArtManager.cpp

```diff
@@ -29,13 +29,14 @@ const QIcon& ArtManager::GetIcon(const QString& name) {
 
 const QBrush& ArtManager::GetTransparenyBrush() { return transparenyBrush; }
 
-const QPixmap& ArtManager::GetCachedPixmap(const QString& name) {
-  QPixmap pm;
+bool ArtManager::GetCachedPixmap(const QString& name, QPixmap &pm) {
+//  QPixmap pm;
   if (!QPixmapCache::find(name, &pm)) {
-    pm.load(name);
-    QPixmapCache::insert(name, pm);
+    if (!pm.load(name)) return false;
+     QPixmapCache::insert(name, pm);
```

Comment:
`return QPixmapCache::insert(name, pm);`

This line might refuse to add a pixmap to the cache and return false.